### PR TITLE
Future flatten method

### PR DESCRIPTION
**Pull request by oshai**

### CODE REVIEW

File: ob1k-concurrent/src/test/java/com/outbrain/ob1k/concurrent/ComposableFutureTest.java [hunk 1]
```diff
@@ -169,6 +170,35 @@ public void testDelayedFuture() throws Exception {
...
+
+  @Test
+  public void testFlattenFailure() throws Exception {
+    final ComposableFuture<String> errorFuture = fromError(new IllegalStateException("errr"));
+    final ComposableFuture<ComposableFuture<String>> errorNestedFuture = fromValue(errorFuture);
+    final ComposableFuture<String> errorFlatten = flatten(errorNestedFuture);
+
+    Throwable cause = getFutureCause(errorFuture);
+    Throwable causeFlatten = getFutureCause(errorFlatten);
+    assertEquals("flatten and original should have same value", cause, causeFlatten);
```
Comment (marenzo) [inline ob1k-concurrent/src/test/java/com/outbrain/ob1k/concurrent/ComposableFutureTest.java]: no really need in this helper method, I'd just change @Test to expect IllegalStateException and do errorFlatten.get();
end of test

try to keep it simple and clean :)

File: ob1k-concurrent/src/main/java/com/outbrain/ob1k/concurrent/ComposableFutures.java [hunk 2]
```diff
@@ -164,6 +164,18 @@ private ComposableFutures() {
     return result;
   }
 
+  /**
+   * Continues a future with a handler that will be called only if the original future resulted with success
```
Comment (marenzo) [inline ob1k-concurrent/src/main/java/com/outbrain/ob1k/concurrent/ComposableFutures.java]: description is unclear.
describe here that the purpose of this method is to flatten a nested future. there's no handler.

for example, Scala's description:

  /** Creates a new future with one level of nesting flattened, this method is equivalent
   *  to `flatMap(identity)`.

File: ob1k-concurrent/src/test/java/com/outbrain/ob1k/concurrent/ComposableFutureTest.java (hunk 1 above)
Comment (oshai) [reply ob1k-concurrent/src/test/java/com/outbrain/ob1k/concurrent/ComposableFutureTest.java]: Thanks, will change that.

File: ob1k-concurrent/src/main/java/com/outbrain/ob1k/concurrent/ComposableFutures.java (hunk 2 above)
Comment (oshai) [reply ob1k-concurrent/src/main/java/com/outbrain/ob1k/concurrent/ComposableFutures.java]: Will change that as well, Thanks for the feedback!

**Pull request merged**